plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchMetadata.java

```diff
@@ -660,6 +688,174 @@ private static boolean isPassthroughQuery(OpenSearchTableHandle table)
+            default -> false;
+        };
+    }
+
+    private static boolean isSupportedForPushdown(ConnectorExpression connectorExpression)
+    {
+        if (connectorExpression instanceof Variable) {
+            return true;
+        }
+        if (connectorExpression instanceof FieldDereference fieldDereference) {
+            RowType rowType = (RowType) fieldDereference.getTarget().getType();
+            RowType.Field field = rowType.getFields().get(fieldDereference.getField());
+            if (field.getName().isEmpty()) {
+                return false;
+            }
+            return !field.getName().get().contains(".");
+        }
+        return false;
+    }
+
+    @VisibleForTesting
+    static OpenSearchColumnHandle projectColumn(OpenSearchColumnHandle baseColumn, List<Integer> indices, Type projectedColumnType)
+    {
+        if (indices.isEmpty()) {
+            return baseColumn;
+        }
+        ImmutableList.Builder<String> dereferencePath = ImmutableList.builder();
+        dereferencePath.addAll(baseColumn.dereferencePath());
+
+        DecoderDescriptor decoderDescriptor = baseColumn.decoderDescriptor();
+        Type type = baseColumn.type();
+
+        for (int index : indices) {
+            checkArgument(type instanceof RowType, "type should be Row type");
+            RowType rowType = (RowType) type;
+            RowType.Field field = rowType.getFields().get(index);
+            dereferencePath.add(field.getName()
+                    .orElseThrow(() -> new TrinoException(NOT_SUPPORTED, "ROW type does not have field names declared: " + rowType)));
+            type = field.getType();
+
+            checkArgument(decoderDescriptor instanceof RowDecoder.Descriptor, "decoderDescriptor should be RowDecoder.Descriptor type");
+            decoderDescriptor = ((RowDecoder.Descriptor) decoderDescriptor).getFields().get(index).getDescriptor();
+        }
+
+        return new OpenSearchColumnHandle(
+                baseColumn.baseName(),
+                dereferencePath.build(),
+                projectedColumnType,
+                decoderDescriptor,
+                supportsPredicates(projectedColumnType));
+    }
+
+    @VisibleForTesting
+    static boolean supportsPredicates(Type type)
+    {
+        return switch (type) {
+            case TimestampType ignore -> true;
+            case BooleanType ignore -> true;
+            case TinyintType ignore -> true;
+            case SmallintType ignore -> true;
+            case IntegerType ignore -> true;
+            case BigintType ignore -> true;
+            case DoubleType ignore -> true;
+            case RealType ignore -> true;
+            default -> false; // VarcharType returns false as `text` and `keyword` native types results into to VarcharType. Further, `text` doesn't support pushdown as per `supportsPredicates(IndexMetadata.Field)`.
```

Comment:
Nit: this line is too long. Maybe put the comment in a separate line, before the `default` case?